qiita_db/metadata_template/prep_template.py

```diff
@@ -244,11 +265,19 @@ def delete(cls, id_):
                                                           cls._id_column),
             (id_,))
 
+        # Remove the rows from prep_template_edit
+        conn_handler.execute(
+            "DELETE FROM qiita.{0} where {1} = %s".format(cls._log_table,
+                                                          cls._id_column),
+            (id_,))
+
         # Remove the row from prep_template
         conn_handler.execute(
             "DELETE FROM qiita.prep_template where "
             "{0} = %s".format(cls._id_column), (id_,))
 
+        cls._update_analyses(id_)
```

Comment:
I think this line is not needed. A prep template cannot be deleted if it's data has been used in an analysis.